packages/react/src/components/ui/WalletConnectModal.tsx

```diff
@@ -12,40 +13,41 @@ const IOS_KEPLR_MOBILE_URL = 'itms-apps://itunes.apple.com/app/1567851089'
 export interface WalletConnectModalProps extends BaseModalProps {
   uri?: string
   reset: () => void
+  deeplinkFormats?: DeeplinkFormats
 }
 
-export const WalletConnectModal: FunctionComponent<WalletConnectModalProps> = ({
+export const WalletConnectModal = ({
   isOpen,
   uri,
   classNames,
   reset,
+  deeplinkFormats,
   ...props
-}) => {
+}: WalletConnectModalProps) => {
   const isMobile = useMemo(() => checkIsMobile(), [])
   const isAndroid = useMemo(() => checkIsAndroid(), [])
 
   // Defined if isMobile is true.
-  const navigateToAppURL = useMemo(
+  const deeplinkUrl = useMemo(
     () =>
-      isMobile
+      isMobile && deeplinkFormats
         ? isAndroid
-          ? `intent://wcV1?${uri}#Intent;package=com.chainapsis.keplr;scheme=keplrwallet;end;`
-          : `keplrwallet://wcV1?${uri}`
+          ? deeplinkFormats.android.replace('{{uri}}', uri)
+          : deeplinkFormats.ios.replace('{{uri}}', uri)
         : undefined,
     [isMobile, isAndroid, uri]
   )
 
   // Open app if mobile URL is available.
   useEffect(() => {
-    if (!isOpen || !navigateToAppURL) return
+    if (!isOpen || !deeplinkUrl) return
 
-    // Slight delay so they can read the modal.
     const timeout = setTimeout(() => {
-      window.location.href = navigateToAppURL
-    }, 2000)
+      window.location.href = deeplinkUrl
+    }, 100)
```

Comment:
we no longer want a delay to read the modal?

Reply:
I decided it was more annoying than helpful, as most people will be comfortable clicking an `Open Keplr Mobile` prompt after pressing the `Connect Wallet` button, especially after reading the modal once. For those who have already used it, a 2 second delay is just lag.